metrics.go

```diff
@@ -5,7 +5,16 @@ import (
 	"time"
 )
 
+type Label struct {
+	Name  string
+	Value string
+}
+
 func (m *Metrics) SetGauge(key []string, val float32) {
+	m.SetGaugeWithLabels(key, val, nil)
+}
+
+func (m *Metrics) SetGaugeWithLabels(key []string, val float32, labels []Label) {
```

Comment:
That's a good idea, I can add an option to add them as labels instead of prepending them to the key